crypto/mem.c

```diff
@@ -226,6 +226,11 @@ void *CRYPTO_zalloc(size_t num, const char *file, int line)
     return ret;
 }
 
+void *CRYPTO_aligned_alloc(size_t num, size_t alignment, const char *file, int line)
```

Comment:
malloc(3) guarantees alignment.  Why do we need this in addition?

Reply:
It guarantees alignment to a multiple of 16 bytes on 64 bit systems (or 8 bytes on non 64 bit systems), To optimize, its desireable to put the entire neighborhood on a single cacheline, which on most systems is 64 bytes.  If we don't align to 64 bytes we risk getting a cache miss part way  through traversing a neighborhood